bliss/simulator/prior.py

```diff
@@ -139,27 +169,96 @@ def _sample_n_galaxies_and_stars(self, n_sources):
 
         return galaxy_bools, star_bools
 
-    def _sample_star_fluxes(self):
-        latent_dims = (self.batch_size, self.n_tiles_h, self.n_tiles_w, self.max_sources, 1)
-        return self._draw_truncated_pareto(
-            self.star_flux_alpha, self.star_flux_min, self.star_flux_max, latent_dims
+    def _sample_star_fluxes(self, star_ratios):
+        latent_dims = (self.batch_size, self.n_tiles_h, self.n_tiles_w, self.max_sources)
+        r_flux = self._draw_truncated_pareto(
+            self.galaxy_alpha, self.galaxy_flux_min, self.galaxy_flux_max, latent_dims
+        )
+        total_flux = self.multiflux(star_ratios, r_flux)
+
+        # select specified bands
+        bands = np.array(self.bands)
+        return total_flux[..., bands]
+
+    def multiflux(self, rcf_ratios, flux_base) -> Tensor:
+        """Generate flux values for remaining bands based on draw."""
+        total_flux = torch.zeros(
+            self.batch_size, self.n_tiles_h, self.n_tiles_w, self.max_sources, 5
         )
 
+        for b, obj_ratios in enumerate(rcf_ratios):
+            for h, w, s, bnd in itertools.product(  # noqa: WPS352
+                range(self.n_tiles_h),
+                range(self.n_tiles_w),
+                range(self.max_sources),
+                range(self.max_bands),
+            ):
+                # sample a light source
+                flux_sample = obj_ratios[np.random.randint(0, len(obj_ratios) - 1)]
+                total_flux[b, h, w, s, bnd] = flux_sample[bnd] * flux_base[b, h, w, s]
+
+        return total_flux
+
+    def _sample_ratios(self, sdss_fields):
```

Comment:
Is this function sampling something, or just loading a bunch of data? sampling = random. But it looks like this function is just loading. If so, it'd be good to name it something like `_load_color_distribution`